initialize.py

```diff
@@ -117,79 +115,83 @@ def parse_config(path_cfile):
                 config_params["global_toa_path"] = sline[1].rstrip().lstrip()
             if attr == "redundant_products":
                 config_params["red_prod"] = sline[1].rstrip().lstrip(' ').split(',')
+            if attr == "venv":
+                config_params["venv"] = sline[1].rstrip().lstrip()
+            if attr == "soft_dir":
+                config_params["soft_dir"] = sline[1].rstrip().lstrip()
 
     cfile.close()
-    
+
     return config_params
 
 
-def setup_logging(path,verbose,file_log):
```

Comment:
I'll be honest, I have never fully understood this function and have largely left it alone. If this is the code behind the log files of yours that I inspected when verifying the operation of your code on NT, then I am satisfied with its behaviour - it produces cleaner output than the current logs, matches the key messages 1:1, and seems relatively streamlined.

Reply:
It is, so I am glad you approve. I wasn't quite sure what the other one did so hopefully my new function is a bit clearer